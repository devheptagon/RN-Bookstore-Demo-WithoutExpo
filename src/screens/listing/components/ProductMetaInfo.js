import React from "react";
import PropTypes from "prop-types";
import { Text, StyleSheet } from "react-native";
import Rating from "../../components/Rating";
import { formatMoney } from "../../../utils/helper";
import Theme from "../../../constants/Theme";

export const ProductMetaInfo = props => {
  const { volumeInfo, saleInfo, description, isbn } = props.dataItem;
  const { title, subtitle, authors, publishedDate, averageRating } = volumeInfo;
  const { listPrice } = saleInfo;
  const amount = listPrice ? listPrice.amount : 1;
  const { isDetailed } = props;
  return (
    <React.Fragment>
      <Text numberOfLines={isDetailed ? 5 : 1} style={styles.Title}>
        {title}
      </Text>
      <Text numberOfLines={isDetailed ? 5 : 1} style={styles.Subtitle}>
        {subtitle}
      </Text>
      <Text numberOfLines={isDetailed ? 5 : 1} style={styles.Author}>
        {authors && authors[0]} - {publishedDate}
      </Text>
      {isDetailed && (
        <React.Fragment>
          <Text style={styles.Description}>{description}</Text>
          <Text style={styles.Isbn}>
            {isbn && isbn.type + ": " + isbn.identifier}
          </Text>
        </React.Fragment>
      )}
      <Rating value={averageRating} />
      <Text style={styles.Amount}>{formatMoney(amount)}</Text>
    </React.Fragment>
  );
};

const styles = StyleSheet.create({
  Title: {
    fontSize: Theme.General.Text.FontBig,
    fontWeight: Theme.General.Text.FontBold
  },
  Subtitle: {
    fontSize: Theme.General.Text.FontRegular,
    color: Theme.ProductList.Item.AlternateTextColor1
  },
  Author: {
    fontSize: Theme.General.Text.FontSmall,
    color: Theme.ProductList.Item.AlternateTextColor2
  },
  Amount: {
    fontSize: Theme.General.Text.FontBig,
    fontWeight: Theme.General.Text.FontBold,
    color: Theme.General.Text.AmountColor
  },
  Description: {
    fontSize: Theme.General.Text.FontSmall
  },
  Isbn: {
    fontWeight: Theme.General.Text.FontBold
  }
});

ProductMetaInfo.propTypes = {
  volumeInfo: PropTypes.object,
  saleInfo: PropTypes.object,
  description: PropTypes.string,
  isbn: PropTypes.object
};

export default ProductMetaInfo;
