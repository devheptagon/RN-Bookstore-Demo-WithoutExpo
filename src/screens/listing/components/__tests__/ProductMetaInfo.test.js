import React from 'react';
import {ProductMetaInfo} from '../ProductMetaInfo';
import renderer from 'react-test-renderer';
import { FormattedProvider } from 'react-native-globalize';
import Config from '../../../../config/appConfig'

test('ProductMetaInfo renders correctly', () => {
  const dataItem = {      
    "id": "GqTwl7-WZtUC",    
    "volumeInfo": {
     "title": "Google Hacks",
     "authors": [
      "Tara Calishain"
     ],     
     "publishedDate": "2003",
     "description": "Explains how to take advantage of Google's user interface, discussing how to filter results, use Google's special services, integrate Google applications into a Web site or Weblog, write information retrieval programs, and play games.",
     "industryIdentifiers": [
      {
       "type": "ISBN_10",
       "identifier": "0596004478"
      }
     ],     
     "averageRating": 4.0,     
     "imageLinks": {
      "smallThumbnail": "http://books.google.com/books/content?id=GqTwl7-WZtUC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api",
      "thumbnail": "http://books.google.com/books/content?id=GqTwl7-WZtUC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
     },     
    },
      "saleInfo": {
     "listPrice": {
      "amount": 18.99,
      "currencyCode": "GBP"
     }
   }
  }
  const component = <FormattedProvider locale={Config.locale} currency={Config.currency}>
    <ProductMetaInfo dataItem={dataItem} />
  </FormattedProvider>
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});