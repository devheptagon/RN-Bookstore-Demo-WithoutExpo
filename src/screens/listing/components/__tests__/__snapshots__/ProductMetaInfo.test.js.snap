// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductMetaInfo renders correctly 1`] = `
Array [
  <Text
    style={
      Object {
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    Google Hacks
  </Text>,
  <Text
    style={
      Object {
        "color": "green",
        "fontSize": 12,
      }
    }
  />,
  <Text
    style={
      Object {
        "color": "#333",
        "fontSize": 11,
      }
    }
  >
    Tara Calishain
     - 
    2003
  </Text>,
  <View
    pointerEvents="auto"
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 5,
        "marginTop": 5,
        "width": "30%",
      }
    }
  >
    <View
      style={Object {}}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={Object {}}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={Object {}}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={Object {}}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={Object {}}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>,
  <Text
    style={
      Object {
        "color": "green",
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  >
    <Text
      accessibilityLabel=""
      accessible={true}
      adjustsFontSizeToFit={false}
      allowFontScaling={true}
      style={null}
    >
      Â£18.99
    </Text>
  </Text>,
]
`;
